public/app/features/gops/configuration-tracker/components/ConfigureIRM.tsx

```diff
@@ -58,7 +59,11 @@ export function ConfigureIRM() {
     });
   }, [dataSourceConfigurationData.dataSourceCompatibleWithAlerting]);
 
-  const [essentialsOpen, setEssentialsOpen] = useState(false);
+  // query param 'essentials' is used to open essentials drawer when landing on the page
+  const [queryParams] = useURLSearchParams();
+  const essentialsIsOpenFromUrl = queryParams.get('essentials') === 'open';
+
+  const [essentialsOpen, setEssentialsOpen] = useState(essentialsIsOpenFromUrl);
```

Comment:
How about not keeping the local state at all and instead rely only on the query parameter value?
AFAIK `useURLSearchParams` also returns a function which gives you the way to set the value of query params

Reply:
this state is needed to handle opening the 'generic' `ConfigCard`. Maybe we could keep this state in here , but I believe this way is more clear and we don't depend on the existence of this query parameter.